woocommerce/woocommerce
# DateTimePickerControl: Only call onChange when the date actually changes

### All Submissions:






### Changes proposed in this Pull Request:



This PR fixes the change handling to only call onChange when the date actually changes.

In addition, the change-related code has been refactored to reduce duplication, needless re-parsing, and increase readability.

Needed for #34538






### How to test the changes in this Pull Request:



1. Run unit tests: `pnpm --filter=@woocommerce/components run test -- src/date-time-picker-control`
2. Interact with Storybook and make sure DateTimePickerControl stories work as expected.
    - View the `onChange` entries in the Storybook "Actions" tab to confirm correct behavior.



### Other information:




### FOR PR REVIEWER ONLY:
